libsql-sqlite3/src/wal.h

```diff
@@ -190,11 +163,37 @@ struct WalIndexHdr {
   unsigned int aCksum[2];                  /* Checksum over all prior fields */
 };
 
+typedef struct libsql_create_wal libsql_create_wal;
+struct libsql_create_wal {
+  /* True if the implementation relies on shared memory routines (e.g. locks) */
+  int bUsesShm;
+
+  /*
+  ** This optional callback gets called before the main database file which owns
+  ** the WAL file is open. It is a good place for initialization routines, as WAL
+  ** is otherwise open lazily.
+  */
+  int (*xPreMainDbOpen)(void* self, const char *main_db_path, int main_db_path_len);
```

Comment:
Side note: since we overhaul the interface, we should consider removing this thing altogether, it makes very little sense. I thought I need it long time ago when `bottomless` was developed as a loadable module, but even then, it makes 10000x more sense to just expose an initialization method that users are supposed to call before a database is open, not a weird WAL methods hook. So if we decide on the shape of virtual WAL v2, it should also include nuking this xPreMainDbOpen hook.